service/history/workflow/delete_manager.go

```diff
@@ -209,8 +202,17 @@ func (m *DeleteManagerImpl) deleteWorkflowExecutionInternal(
 		currentBranchToken = nil
 	}
 
+	var startTime *time.Time
+	var closeTime *time.Time
+
 	completionEvent, err := ms.GetCompletionEvent(ctx)
-	if err != nil {
+	switch {
+	case err == ErrMissingWorkflowCompletionEvent:
```

Comment:
I think we should call `GetCompletionEvent` or `GetStartEvent` based on the workflow state instead of handling the error. This way we can still catch the unexpected case where a workflow is closed by no close event.

Reply:
make sense